### DISCUSSION TITLE
When browsing a lot of photos, it starts to bog down especially when using phone or tablet.

### DISCUSSION THREAD

**Discussion by hiro5id** [photoprism/photoprism#2028] [Q&A] [answered]
One of my favourite things to do when I have close family over is to project photoPrism from my iPad to the TV using Apple TV, and just let it run the slide show through all of our family photos.  We let it run in the background when we have family over for dinner or some other occasion.
We have a lot of photos, over 200,000.

The slideshow works fine for the first 15-20 minutes, but as time goes on and it goes further and further into the photos, PhotoPrism starts to get bogged down in the browser, eventually to the point that it becomes unresponsive and eventually crashes.

I am wondering if this is a bug.  Perhaps lacking cleanup in the browser DOM as you scroll through, maybe it should be cleaning up things that are not in view anymore and removing them from the DOM.  

I am only speculating of course.  I know with Angular it is very important to cleanup Observables for example or they become memory leaks.

There is definitely an issue with it getting bogged down.

I can even reproduced it just by opening the "search" tab, that shows all my photos, and then just keep scrolling down, eventually it starts to get slower and slower, to the point that the browser gives up.

Is this a worthy thing to open an issue on?

**Comment by graciousgrey** [comment, marked as answer]
PhotoPrism's current views are optimized for searching and filtering search results, not for endless scrolling.
What you need for endless scrolling is a timeline view that is optimized for that use case. It's already on our roadmap: https://github.com/photoprism/photoprism/issues/152 :)
